--- quicklzpy.c	2025-02-09 19:43:13
+++ quicklzpy.c	2025-02-09 19:42:50
@@ -133,10 +133,10 @@
     return -1;
 }
 
-int qlz_compare(PyObject *self, PyObject *other)
+PyObject* qlz_compare(PyObject *self, PyObject *other, int op)
 {
     // 'is a' comparison
-    return self == other;
+    return PyBool_FromLong(self == other);
 }
 
 int qlz_c_print(PyObject *self, FILE *fp, int flags)
@@ -188,7 +188,7 @@
   sizeof(qlz_state_compress_Type),/* int tp_basicsize; */
   0,                          /* int tp_itemsize;       /* not used much */
   (destructor)qlz_state_compress_dealloc, /* destructor tp_dealloc; */
-  qlz_c_print,                /* printfunc  tp_print;   */
+  0,                /* printfunc  tp_print;   */
   qlz_py_getattr,             /* getattrfunc  tp_getattr; /* __getattr__ */
   qlz_py_setattr,             /* setattrfunc  tp_setattr;  /* __setattr__ */
   qlz_compare,                          /* cmpfunc  tp_compare;  /* __cmp__ */
@@ -232,7 +232,7 @@
   sizeof(qlz_state_decompress_Type),/* int tp_basicsize; */
   0,                          /* int tp_itemsize;       /* not used much */
   (destructor)qlz_state_decompress_dealloc,/* destructor tp_dealloc; */
-  qlz_d_print,                /* printfunc  tp_print;   */
+  0,                /* printfunc  tp_print;   */
   qlz_py_getattr,             /* getattrfunc  tp_getattr; /* __getattr__ */
   qlz_py_setattr,             /* setattrfunc  tp_setattr;  /* __setattr__ */
   qlz_compare,                          /* cmpfunc  tp_compare;  /* __cmp__ */
@@ -450,7 +450,7 @@
 
 #define INITERROR return NULL
 
-PyObject *
+PyMODINIT_FUNC
 PyInit_quicklz(void)
 #else

--- setup.py	2015-03-22 08:54:08
+++ setup.py	2025-02-09 19:50:18
@@ -17,6 +17,7 @@
                "-Wall",
                "-W",
                "-Wundef",
+               "-fPIC", "-fvisibility=default","-undefined", "dynamic_lookup",
 #           try fortification
 #            "-DFORTIFY_SOURCE=2", "-fstack-protector",
 #           try hard CPU optimization